app/services/chf/create_dzi_service.rb

```diff
@@ -0,0 +1,263 @@
+require 'concurrent'
+require 'aws-sdk'
+
+
+module CHF
+  # Needs 'vips' installed.
+  # Will overwrite if it's already there on S3, unless called with `lazy:true`, which
+  # is mainly useful for speed.
+  #
+  # s3 bucket needs CORS turned on! http://docs.aws.amazon.com/AmazonS3/latest/user-guide/add-cors-configuration.html
+  #
+  # For performance, pass in just a file_id, it's all we need, we try to avoid lookups,
+  # but if you can pass in a checksum that you got cheaply, you'll avoid us having to get
+  # it expensively.
+  #
+  # objects are stored in S3 with a key created from both file_id and checksum, to make them
+  # self-cache-busting if the datastream at a file_id changes (versioning?). Does
+  # make the s3 keys kind of non-human-readable, but I think that's okay.
+  #
+  # Uses a bunch of Chf::Env settings:
+  # required:
+  #  * aws_access_key_id
+  #  * aws_secret_access_key
+  # optional (cause they have defaults):
+  #  * dzi_job_tmp_dir
+  #  * dzi_s3_bucket
+  #  * dzi_s3_bucket_region
+  #
+  # Seek rake chf:dzi for tools for managing the S3 bucket contents.
+  #
+  # MAYBE would we better off using actual libvips bindings at https://github.com/jcupitt/ruby-vips
+  #   instead of shell out? I think it probably doesn't matter.
+  #
+  # TODO what do we want to do in dev? create dzi's to public/? No DZIs?
+  #
+  # TODO some cleverer concurrency stuff if two of these jobs try acting at the same
+  # time, keep the out of using each others files, or let them actually share/wait
+  # on each other files.
+  class CreateDziService
+    WORKING_DIR = CHF::Env.lookup(:dzi_job_tmp_dir)
+    UPLOAD_THREADS = 128
+
+    class_attribute :vips_command
+    self.vips_command = "vips"
+
+    class_attribute :jpeg_quality
+    self.jpeg_quality = "85"
+
+    class_attribute :suppress_vips_stderr
+    self.suppress_vips_stderr = true
+
+    class_attribute :cache_control
+    self.cache_control = "max-age=31536000"
```

Comment:
That is literally copied from Amazon's suggestion. It's one year. will add comment.